test/e2e/deployment.go

```diff
@@ -226,8 +235,196 @@ func testRollingUpdateDeploymentEvents(f *Framework) {
 			Name: deploymentName,
 		},
 		Spec: extensions.DeploymentSpec{
-			Replicas:       1,
-			Selector:       deploymentPodLabels,
+			Replicas: 1,
+			Selector: deploymentPodLabels,
+			Strategy: extensions.DeploymentStrategy{
+				Type: extensions.RollingUpdateDeploymentStrategyType,
+			},
+			UniqueLabelKey: "deployment.kubernetes.io/podTemplateHash",
```

Comment:
No constant found. We can refactor it as a constant. I am happy to help fix this.